pkg/arvo/app/chat-cli.hoon

```diff
@@ -0,0 +1,1111 @@
+::  chat-cli: cli chat client using chat-store and friends
+::
+::    pulls all known messages into a single stream.
+::    type ;help for usage instructions.
+::
+::    note that while the chat-store only cares about paths,
+::    we mostly deal with [ship path] (aka target) here.
+::    when sending messages (through the chat hook),
+::    we concat the ship onto the head of the path,
+::    and trust it to take care of the rest.
+::
+/-  sole-sur=sole, *chat-store, *chat-view, *chat-hook
+/+  sole-lib=sole
+/=  seed  /~  !>(.)
+::
+|%
++$  state
+  $:  grams=(list mail)                             ::  all messages
+      known=(set [target serial])                   ::  known message lookup
+      count=@ud                                     ::  (lent grams)
+      bound=(map target char)                       ::  bound circle glyphs
+      binds=(jug char target)                       ::  circle glyph lookup
+      audience=(set target)                         ::  active targets
+      settings=(set term)                           ::  frontend flags
+      width=@ud                                     ::  display width
+      timez=(pair ? @ud)                            ::  timezone adjustment
+      cli=[=bone state=sole-share:sole-sur]         ::  console id & state
+  ==
+::
++$  mail  [source=target envelope]
++$  target  [=ship =path]
+::
++$  glyph  char
+++  glyphs  "!@#$%^&()-=_+[]\{};'\\:\",.<>?"
+::
++$  command
+  $%  [%target (set target)]                        ::  set messaging target
+      [%say (list letter)]                          ::  send message
```

Comment:
Why is this a list? Can we name this command %message or something that maps closer to the semantics of %chat-store?

Reply:
Sure thing about the name. This is a list because talk had, and now chat-cli has, a convention of being able to send separate messages in a single command by splitting with `•`. I... could take that out, I suppose? This also future-proofs for more complex message sending semantics (like those of previous web chat), but something tells me we'll never ever get those.

Reply:
please take it out. seems unnecessarily complex. up to you, but if it were me I'd remove.